src/decoder/stream.rs

```diff
@@ -911,6 +912,43 @@ impl StreamingDecoder {
         }
     }
 
+    fn parse_iccp(&mut self) -> Result<Decoded, DecodingError> {
+        if self.have_idat {
+            Err(DecodingError::Format(
+                FormatErrorInner::AfterIdat { kind: chunk::iCCP }.into(),
+            ))
+        } else {
+            let mut buf = &self.current_chunk.raw_bytes[..];
+
+            // read profile name
+            let _: u8 = buf.read_be()?;
+            for _ in 1..80 {
+                let raw: u8 = buf.read_be()?;
+                if raw == 0 {
+                    break;
+                }
+            }
+
+            match buf.read_be()? {
+                // compression method
+                0u8 => (),
+                n => {
+                    return Err(DecodingError::Format(
+                        FormatErrorInner::UnknownCompressionMethod(n).into(),
+                    ))
+                }
+            }
+
+            let mut profile = Vec::new();
+            let mut inflater = ZlibStream::new();
+            inflater.decompress(buf, &mut profile)?;
+            inflater.finish_compressed_chunks(&mut profile)?;
+
+            self.info.as_mut().unwrap().icc_profile = Some(profile);
+            Ok(Decoded::Nothing)
```

Comment:
Is there a way we can impose a maximum size for the ICC profile to prevent a malicious file from causing us to allocate an unreasonable amount of memory here?

Reply:
PR would look good to me with this resolved.

Reply:
How is this problem handle elsewhere? Is there a way to cap the size that inflater will make the Vec?

Reply:
libpng uses PNG_USER_CHUNK_MALLOC_MAX which defaults to 8000000

Reply:
The maximum should be capped at most at `Limits::bytes` from the field in `Decoder`. (It defaults to 64Mb but that's not a constant, it can be configured before decoding). Maybe it should be less because it's supposed to be somewhat of a total maximum.

Reply:
What about doing the actual capping? It seems like finish_compressed_chunks() will happily keep allocating as much as the the zlib stream wants.

Reply:
As stated in the comment there, yes, we may want to fix that at some point. But that's less critical than the main decompression. Open a separate PR for it, if you can fix it?

Reply:
Well, yes, that means you can't control the final size if everything is decoded in a single call to `decompress`. That's suboptimal indeed. To clarify, you can however call `decompress` with an empty slice so that each call appends at most `CHUNCK_BUFFER_SIZE` bytes of data to `profile`. Repeat until you're reasonably sure that there is no remaining input data and then call finish.. That's not optimal but should be quite good for a start?